### DISCUSSION TITLE
Multiobjective Surrogate Based Optimization

### DISCUSSION THREAD

**Discussion by kasia129** [snl-dakota/dakota#28] [Q&A]
Hello, 

Do you have any recommendations for setting up a Multiobjective Surrogate Based Optimization with Dakota? Currently I have a three-stage setup, for a single objective problem:

1) Sample the space with `fsu_quasi_mc	halton`
2) Generate in-fill points in the design space with` efficient_global `method
3) Hybrid sequential optimisation, with `coliny_ea` and `coliny_pattern_search`

This setup worked well so far, considering the high cost of each response evaluation (Computational Fluid Dynamics).

I would like to extend this case to a multiobjective scenario, replacing method in step 3 with `moga` (Multi-objective Genetic Algorithm). In that case, is it possible to use `efficient_global` method for multiple responses case? Or do you recommend a different setup?

Thanks
Kasia

**Comment by jadamstephens** [comment]
It depends on what you mean by multiobjective. The `efficient_global` method is a single-objective optimizer. If you have more than one objective, Dakota will combine them by doing a weighted sum. You can specify the weights. The underlying Gaussian process that `efficient_global` constructs will be (I believe..) for the weighted sum.

`moga` is a true multiobjective optimizer that aims to identify the Pareto surface for the problem. You have a couple of options to make use of surrogates in this case.

One is to use the [surrogate_based_global](https://dakota.sandia.gov/sites/default/files/docs/6.17.0-release/user-html/usingdakota/reference/method-surrogate_based_global.html) method. It works a tiny bit like `efficient_global`, in that solutions provided by the optimizer are evaluated using the truth model, and the surrogates are retrained.

Another option is to generate training data, build surrogates once, and run `moga` on them. This approach is illustrated in these examples.

* [Build surrogates from pre-computed data.](https://github.com/snl-dakota/dakota-examples/tree/master/official/surrogates/imported)
* [Build surrogates using data obtained in the same Dakota study.](https://github.com/snl-dakota/dakota-examples/tree/master/official/surrogates/dace)

The top level method in the examples is `sampling` and not `moga`, but the structure of the input file would be the same.
